tools/RAiDER/getStationDelays.py

```diff
@@ -0,0 +1,184 @@
+    #import functions
+    import gzip, zipfile, gzip, requests, io
+
+    #Refer to the following sites to interpret stationFile variable names:
+    #ftp://igs.org/pub/data/format/sinex_tropo.txt
+    #http://geodesy.unr.edu/gps_timeseries/README_trop2.txt
+
+    # sort through station zip files
+    allstationTarfiles=[]
+    # if URL
+    if stationFile.startswith('http'):
+        r = requests.get(stationFile)
+        ziprepo = zipfile.ZipFile(io.BytesIO(r.content))
+    # if downloaded file
+    else :
+        ziprepo = zipfile.ZipFile(stationFile)
+    # iterate through tarfiles
+    stationTarlist = ziprepo.namelist()
+    stationTarlist.sort()
+    for j in stationTarlist:
+        f = gzip.open(ziprepo.open(j),'rb')
+        # get the date of the file
+        time, yearFromFile, doyFromFile = getDate(j)
+        # initialize variables
+        d, ngrad, egrad, timesList, Sig = [], [], [],[],[]
+        flag = False
+        for line in f.readlines():
+            try:
+                line = line.decode('utf-8')
+            except:
+                line = line.decode('latin-1')
+            if flag:
+                # Do not attempt to read header
+                if 'SITE' in line:
+                   continue
+                # Attempt to read data
+                try:
+                    #units: mm, mm, mm, deg, deg, deg, deg, mm, mm, K
+                    trotot, trototSD, trwet, tgntot, tgntotSD, tgetot, tgetotSD, wvapor, wvaporSD, mtemp = \
+                        [float(t) for t in line.split()[2:]]
+                except:
+                    continue
+                site = line.split()[0]
+                year, doy, seconds = [int(n) for n in line.split()[1].split(':')]
+                # Break iteration if time from line in file does not match date reported in filename
+                if doy != doyFromFile:
+                   print('WARNING: time %s from line in conflict with time %s from file %s, will continue reading next tarfile(s)' \
+                       %(doy,doyFromFile,j))
+                   continue
+                d.append(trotot)
+                ngrad.append(tgntot)
+                egrad.append(tgetot)
+                timesList.append(seconds)
+                Sig.append(trototSD)
+            if 'TROP/SOLUTION' in line:
+                flag = True
+        del f
+        # Break iteration if file contains no data.
+        if d == []:
+            print('WARNING: file %s for station %s is empty, will continue reading next tarfile(s)'%(j, j.split('.')[0]))
+            continue
+
+        # check for missing times
+        true_times = list(range(0,86400, 300))
+        if len(timesList)!=len(true_times):
+           missing = [True if t not in timesList else False for t in true_times]
+           mask = np.array(missing)
+           delay, sig, east_grad, north_grad = [np.full((288,), np.nan)]*4 
+           delay[~mask] = d
+           sig[~mask] = Sig
+           east_grad[~mask] = egrad
```

Comment:
We may want to add a note in the file as per Geoff Blewitt's comment that they have the north gradient and east gradient swapped in their files. From his email: 

"Thanks David.   At least we don't have any data loss, which is the most important thing.

Before I forget to mention it, please note that, due to a formatting error, our tropospheric gradient columns are interchanged.    That is, the column labeled as the East gradient is actually the North, and vice versa.   But you may be only using ZTD in which case, no problem."

If in the future anyone wants to use the gradients we could leave a comment directing them to check with the UNR folks about the proper format.

Reply:
@jlmaurer @sssangha perhaps a broader question would be do we even need to gradient in east, north, up. We leverage the delays themselves nto the gradients. This might just add more overhead to our files that is not used by RAiDER.

Reply:
@dbekaert True but the overhead seems likely to be minimal; as long as we're doing nothing but passing the values; seems like the info it could be useful to some users, but perhaps not.

Reply:
Ok, added note/reference to that formatting error and made sure to flip the order of the initialized variables to account for this. I agree with @jlmaurer that we should just pass instead of omitting altogether